src/components/admin/ChannelConfiguration/ChannelConfiguration.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useAsyncFn } from "react-use";
+import { HeaderButton } from "components/admin/HeaderButton";
+import { Section } from "components/admin/Section";
+import { SectionHeading } from "components/admin/SectionHeading";
+import { SectionTitle } from "components/admin/SectionTitle";
+import { TablePanel } from "components/admin/TablePanel";
+
+import * as adminApi from "api/admin";
+
+import { Channel, MeetingRoomVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { ConfirmationModal } from "../ConfirmationModal/ConfirmationModal";
+
+import { ChannelAddEditModal } from "./ChannelAddEditModal";
+import { ChannelRow } from "./ChannelRow";
+
+interface ChannelConfigurationProps {
+  space: WithId<MeetingRoomVenue>;
+}
+
+export const ChannelConfiguration: React.FC<ChannelConfigurationProps> = ({
+  space,
+}) => {
+  const {
+    isShown: isShownCreateModal,
+    hide: hideCreateModal,
+    show: showCreateModal,
+  } = useShowHide(false);
+
+  const {
+    isShown: isShownDeleteModal,
+    hide: hideDeleteModal,
+    show: showDeleteModal,
+  } = useShowHide(false);
+
+  const [editingIndex, setEditingIndex] = useState<number>();
+
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  useEffect(() => {
+    // The cleanup below is only triggered when a deletion is complete and
+    // the number of channels is reduced because we have received new data
+    // The use of channels.length in the dependencies is important.
+    if (isDeleting) {
+      return () => {
+        setIsDeleting(false);
+        setEditingIndex(undefined);
+      };
+    }
+  }, [space.channels?.length, isDeleting]);
+
+  const saveChannel = async (channelData: Channel) => {
+    await adminApi.upsertChannel(channelData, space.id, editingIndex);
+    setEditingIndex(undefined);
+  };
+
+  const deleteChannel = useAsyncFn(async () => {
```

Comment:
`useAsyncFn` actually handles the loading state quite gracefully. You can see the implementation here: https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md

Ex: const [{loading: isDeleting}, deleteChannel] = useAsyncFn....

Reply:
This is all related to the logic to handle that the data is updated in response to a push from the database rather than being the response from the function.